drivers/dsp0401/dsp0401.c

```diff
@@ -0,0 +1,249 @@
+    { 0x00, 0x00 },  /* 115:   s  */
+    { 0x00, 0x00 },  /* 116:   t  */
+    { 0x00, 0x00 },  /* 117:   u  */
+    { 0x00, 0x00 },  /* 118:   v  */
+    { 0x00, 0x00 },  /* 119:   w  */
+    { 0x00, 0x00 },  /* 120:   x  */
+    { 0x00, 0x00 },  /* 121:   y  */
+    { 0x00, 0x00 },  /* 122:   z  */
+    { 0x85, 0x11 },  /* 123:   {  */
+    { 0x01, 0x01 },  /* 124:   |  */
+    { 0x11, 0x85 },  /* 125:   }  */
+    { 0x00, 0x00 },  /* 126:   ~  */
+    { 0x00, 0x00 },  /* 127:      */
+};
+
+static void _shift_char(dsp0401_t *dev, uint8_t c)
+{
+    /* Unsupported chars not displayed */
+    if ((c < CHAR_MAP_MIN) || (c > CHAR_MAP_MAX)) {
+        c = ' ';
+    }
+
+    for (unsigned p = 0 ; p < 2 ; ++p) {
+        for (unsigned i = 0; i < 8; ++i)  {
+            gpio_write(SDI, !!(dsp0401_charmap[c - CHAR_MAP_MIN][p] & (1 << (7 - i))));
+            gpio_set(SCK);
+            gpio_clear(SCK);
+        }
+    }
+}
+
+static void _latch(dsp0401_t *dev)
+{
+    xtimer_usleep(LATCH_DELAY);
+    gpio_set(LAT);
+    xtimer_usleep(LATCH_DELAY);
+    gpio_clear(LAT);
+    xtimer_usleep(LATCH_DELAY);
+}
+
+int dsp0401_init(dsp0401_t *dev, const dsp0401_params_t *params)
+{
+    dev->params = *params;
+
+    /* Initialize clock pin and set it to low */
+    if (gpio_init(SCK, GPIO_OUT) < 0) {
+        DEBUG("[ERROR] CLK GPIO initialization failed.");
+        return -DSP0401_ERR_CLK_GPIO;
+    }
+    gpio_clear(SCK);
+
+    /* Initialize data pin */
+    if (gpio_init(SDI, GPIO_OUT) < 0) {
+        DEBUG("[ERROR] SDI GPIO initialization failed.");
+        return -DSP0401_ERR_SDI_GPIO;
+    }
+    gpio_clear(SDI);
+
+    /* Initialize latch pin and set it to low */
+    if (gpio_init(LAT, GPIO_OUT) < 0) {
+        DEBUG("[ERROR] LAT GPIO initialization failed.");
+        return -DSP0401_ERR_LAT_GPIO;
+    }
+    gpio_clear(LAT);
+
+    /* Initialize PWM used for LED brightness */
+    if (!pwm_init(PWM_PERIPH, PWM_LEFT, PWM_FREQ, PWM_STEPS)) {
+        DEBUG("[ERROR] PWM initialization failed.");
+        return -DSP0401_ERR_PWM;
+    }
+    pwm_set(PWM_PERIPH, PWM_CHAN, BRIGHTNESS);
+
+    dsp0401_clear_text(dev);
+
+    return DSP0401_OK;
+}
+
+void dsp0401_display_text(dsp0401_t *dev, char *text)
+{
+    unsigned text_len = strlen(text);
+    for (unsigned i = 0 ; i < MOD_COUNT * 4; ++i) {
+        if (i < text_len) {
+           _shift_char(dev, text[i]);
+        } else {
+            _shift_char(dev, ' ');
+        }
+    }
+
+    _latch(dev);
+}
+
+void dsp0401_clear_text(dsp0401_t *dev)
+{
+    /* each module has 4 alphanumeric displays */
+    for (unsigned i = 0; i < MOD_COUNT * 4; ++i) {
+        _shift_char(dev, ' ');
+    }
+    _latch(dev);
+}
+
+void dsp0401_scroll_text(dsp0401_t *dev, char *text, uint16_t delay)
+{
+    dsp0401_clear_text(dev);
+
+    for (unsigned i = 0; i < strlen(text); ++i) {
+        _shift_char(dev, text[i]);
+        _latch(dev);
+        xtimer_usleep((uint32_t)(delay * US_PER_MS));
```

Comment:
Sorry for replying late. The value here is correct according to the documentation. `delay` is unsigned short which is well suited here.